### DISCUSSION TITLE
Workspace command history?

### DISCUSSION THREAD

**Discussion by muellerto** [leo-arch/clifm#310] [General]
Yesterday I worked with two workspaces. I use them like windows or panes for different tasks in different directories with different files.

Doing this I saw that workspaces share their command history, there's only one. And I thought that's not good because I did completely different things in the two workspaces. So I always got command lines from the history making no sense because they came from the other workspace. And whenever I changed the workspace it was still the same with the other command lines.

I thought it would be good to have **a workspace specific command history** when `PrivateWorkspaceSettings` is `true`.

OK, it gets a problem then to save the history because you have then indeed histories. So you need then also workspace specific history files or at least sections in a file. And of course the user could still get unusable command lines when he uses a workspace with an existing history for something else. But this is then in the responsibility of the user.

**Comment by leo-arch** [comment]
Ok. This one is a feature request. It makes sense but it is not that easy. I'll see what I can do.

**Reply by muellerto** [reply]
No need to hurry. Just some thoughts I had.
